pyne/serpent.py

```diff
@@ -281,11 +319,23 @@ def parse_dep(depfile, write_py=False, make_mats=True):
         else:
             new_filename = depfile.name.rpartition('.')[0] + '.py'
         with open(new_filename, 'w') as pyfile:
-            pyfile.write(f)
+            pyfile.write(full_f)
 
     # Execute the adjusted file
     dep = {}
-    exec(f, dep, dep)
+    try:
+        exec(full_f, dep, dep)
+    except ValueError:
+        correct_footer = form_footer(f, serp2=True)
+        corrected_f = header + f + correct_footer
```

Comment:
here you can probably overwrite the `full_f` to reduce the memory footprint.

Reply:
Fixed the second comment, what change are you suggesting for the first?

Reply:
you should be able to see it [here](https://github.com/pyne/pyne/pull/1225/files#r436215045)